jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/SecureRequestCustomizer.java

```diff
@@ -414,29 +316,16 @@ public Object getAttribute(String name)
         public Set<String> getAttributeNameSet()
         {
             Set<String> names = new HashSet<>(super.getAttributeNameSet());
-            if (getX509(_sslSession) != null)
-                names.add(X509_ATTRIBUTE);
-            String sessionAttribute = getSslSessionAttribute();
-            if (!StringUtil.isNotBlank(sessionAttribute))
+
+            if (_sslSessionData.sslSession() != null)
             {
-                names.add(sessionAttribute);
-                if (_sslSessionData != null)
-                {
-                    names.add(getSslSessionDataAttribute());
-                    names.add(CIPHER_SUITE_ATTRIBUTE);
-                    names.add(KEY_SIZE_ATTRIBUTE);
-                    names.add(SSL_SESSION_ID_ATTRIBUTE);
-                    names.add(PEER_CERTIFICATES_ATTRIBUTE);
-                }
+                names.add(SSL_SESSION_ATTRIBUTE);
+                if (getX509(_sslSessionData.sslSession()) != null)
+                    names.add(X509_ATTRIBUTE);
             }
+            names.add(SSL_SESSION_DATA_ATTRIBUTE);
+
             return names;
         }
     }
-
-    /**
-     * Simple bundle of data that is cached in the SSLSession.
-     */
-    public record SslSessionData(String sessionId, String cipherSuite, int keySize, X509Certificate[] peerCertificates)
-    {
-    }
```

Comment:
We are not removing it, we are just moving it.     Should have been done in 12.0.0, so it is a little late and a little breaking.  But it will only affect users that have adopted the core API since 12.0.0

However, if I make the EndPoint.SslSessionData and interface, then we can have a deprecated extended one here that they can use.

Reply:
We're breaking other things, so ok to remove.